### DISCUSSION TITLE
Page numbering

### DISCUSSION THREAD

**Discussion by HazbarAmat** [dompdf/dompdf#2860] [Q&A]
Hi everyone, 
how can I add page numbers to my pdf pages?

**Comment by bsweeney** [comment]
Page numbers, yes: https://github.com/dompdf/dompdf/wiki/Common-Use-Cases#generated-content

Total number of pages is a bit more problematic. See #1636 and https://github.com/dompdf/dompdf/wiki/Usage#canvas

**Reply by HazbarAmat** [reply]
@bsweeney Thank you for your response. I've copied the same code exactly but I do not see page numbers.  Is there anything I should keep in mind that might affect the numbers to appear?

**Reply by bsweeney** [reply]
For the CSS use case there is nothing extra you would need to do.
